### PR TITLE
Add Notifications AKA Toasts

### PR DESCRIPTION
**Pull request by rottabonus**
# Description

Add toasts for application with [react-hot-toast](https://react-hot-toast.com/docs/toast) library. This library pairs well with the rtk-query that is used in the application. Usage: https://redux-toolkit.js.org/rtk-query/api/createApi#onquerystarted

## Why was the change made?

Toasts are needed for better UX

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unittest
- [x] Cypress e2e -tests

# Caveats?

Does this introduce new warnings or are linter rules suppressed? Why? Is only manual testing applicable for this feature? Is there something special that the reviewer should take into account?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/Buttons/variants.ts [hunk 1]
```diff
@@ -87,95 +87,29 @@ export const variants = {
...
-  searchWithBackground: {
-    backgroundImage: `url(${SearchWithBackgroundIcon})`,
-  },
-  send: {
-    backgroundImage: `url(${SendIcon})`,
-  },
-  tooltip: {
-    backgroundImage: `url(${TooltipIcon})`,
-  },
+  archive: `url(${ArchiveIcon})`,
```
Comment (rottabonus) [inline src/components/Buttons/variants.ts:91]: Note: This was not part of the PR per say. I did stumble across here when I was implementing the toast-icons, but in the end I had to work differently with it. ( Because I could not change the colors ).

Anyway I refactored this at some point to make it more readable, and inferring the icon-names from the record and not declaring a union by hand.

File: tsconfig.json [hunk 2]
```diff
@@ -18,5 +18,5 @@
     "target": "es2015",
     "types": ["node", "jest", "@testing-library/jest-dom"]
   },
-  "include": ["src/*", "@types/*"]
+  "include": ["src/*", "@types/*", "**/*.tsx", "**/*./ts"]
```
Comment (rottabonus) [inline tsconfig.json:21]: Fixed some tsconfig, unrelated to this pr

File: src/components/Buttons/variants.ts (hunk 1 above)
Comment (pheis) [reply src/components/Buttons/variants.ts:91]: This pattern smells here.


Why no svg icon component + icon prop? No need to change now, just feels weird to me.

Comment (pheis) [reply src/components/Buttons/variants.ts:91]: svg filters or css filters do the magic of changing colors :P

File: src/components/Icons/Success.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      <path
+        id="Vector 1 (Stroke)"
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M33.9172 13.3746C35.0123 14.0786 35.3293 15.537 34.6254 16.632L23.5133 33.9175C23.0907 34.575 22.3696 34.9802 21.5883 34.9993C20.8069 35.0184 20.0669 34.6489 19.6126 34.0129L13.4392 25.3702C12.6826 24.3109 12.9279 22.8387 13.9872 22.0821C15.0466 21.3254 16.5187 21.5708 17.2753 22.6301L21.4279 28.4438L30.6599 14.0828C31.3638 12.9877 32.8222 12.6707 33.9172 13.3746Z"
+        fill="currentColor"
+      />
+    </g>
+  </svg>
+);
```
Comment (pheis) [inline src/components/Icons/Success.tsx:36]: whyy inline svgs? for colors probably. This is fine. Just hurts me mentally.

File: src/components/Toaster/Toast.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        return {
+          mainColor: palette.blue,
+          Icon: <Success color={'blueDark'} sizeInPx={23} />,
+        };
+      }
+
+      default: {
+        return {
+          mainColor: palette.purple,
+          Icon: <Warning color={'orange'} sizeInPx={23} />,
```
Comment (pheis) [inline src/components/Toaster/Toast.tsx:42]: Duud, why no 24 to support scaling better? Probably cos design but so close to a multiple of 8

File: src/features/Chat/chatPageApi.ts [hunk 5]
```diff
@@ -161,6 +177,24 @@ export const chatApi = createApi({
...
+          toast.success(t(statusUpdateSuccessMessages[status]));
+        } catch (err) {
+          const statusUpdateErrorMessages = {
+            ok: 'chat:notification.restoringError',
+            banned: 'chat:notification.banningError',
+            archived: 'chat:notification.archivingError',
+          } as const;
+          toast.error(t(statusUpdateErrorMessages[status]));
+        }
+      },
```
Comment (pheis) [inline src/features/Chat/chatPageApi.ts:197]: I would move mappings out of try catch just to see the critical flow better.

await in the middle hurt me :D

**Review by pheis — APPROVED** [review]
Proper good!

I have know few magic tricks to fil svgs with colors, I could share them some day!

Too lazy to write them out here and now :/

File: src/features/Chat/chatPageApi.ts (hunk 5 above)
Comment (rottabonus) [reply src/features/Chat/chatPageApi.ts:197]: yeah i get it. i was also thinking about it. but in the other hand I like that all of the relevant things are next to each other.

File: src/components/Toaster/Toast.tsx (hunk 4 above)
Comment (rottabonus) [reply src/components/Toaster/Toast.tsx:42]: hehe I did actually convert lot of the values to the closest multiple of 8 but not this one, will change it

File: src/components/Icons/Success.tsx (hunk 3 above)
Comment (rottabonus) [reply src/components/Icons/Success.tsx:36]: for colors. I think we need to revisit our svg's soon. not very happy about them atm

File: src/components/Buttons/variants.ts (hunk 1 above)
Comment (rottabonus) [reply src/components/Buttons/variants.ts:91]: yeah, pls share your svg magix

File: src/utils/http.ts [hunk 6]
```diff
@@ -1,11 +1,17 @@
 import { Decoder } from 'io-ts/Decoder';
 import { pipe } from 'fp-ts/function';
-import { Either, isRight } from 'fp-ts/lib/Either';
+import { Either, isRight, isLeft } from 'fp-ts/lib/Either';
 
 const parse =
-  <A, B>(model: Decoder<A, B>) =>
-  (data: A) =>
-    model.decode(data);
+  <A, B>(model: Decoder<A, B>, notification?: () => void) =>
```
Comment (pheis) [inline src/utils/http.ts:6]: This is not a good API.

1. callback should becalled something else than "notification". Maybe `onError`?
2. Callback should maybe give error on the callback to the caller:
`onError: (error: E) => void`

Do u really need callback hook here? This could probably be handled on the callee side?

**Review by pheis — APPROVED** [review]
Pls fix callback hook naming at least :)

File: src/utils/http.ts (hunk 6 above)
Comment (pheis) [reply src/utils/http.ts:6]: Also naming ties this to the place this is used
--> Should be defined there then :).

**Review by pheis — CHANGES_REQUESTED** [review]
Fix naming

File: src/components/Icons/Success.tsx (hunk 3 above)
Comment (rottabonus) [reply src/components/Icons/Success.tsx:36]: Created a ticket for our svg thingy: https://trello.com/c/XfSaXzPo/976-improve-svg-handling

**Review by pheis — APPROVED** [review]
